app/src/main/java/com/asfoundation/wallet/analytics/AmplitudeAnalytics.kt

```diff
@@ -0,0 +1,66 @@
+package com.asfoundation.wallet.analytics
+
+import android.content.Context
+import com.amplitude.api.Amplitude
+import com.amplitude.api.Identify
+import com.amplitude.api.TrackingOptions
+import com.asf.wallet.BuildConfig
+import com.asfoundation.wallet.identification.IdsRepository
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import io.reactivex.schedulers.Schedulers
+
+class AmplitudeAnalytics(private val context: Context, private val idsRepository: IdsRepository) :
+    AnalyticsSetUp {
+
+  private val amplitudeClient = Amplitude.getInstance()
+  private lateinit var entryPoint: String
+
+  override fun setUserId(walletAddress: String) {
+    amplitudeClient.userId = walletAddress
+  }
+
+  override fun setGamificationLevel(level: Int) {
+    val identify = Identify().append(AmplitudeEventLogger.USER_LEVEL, level)
+        .append(AmplitudeEventLogger.APTOIDE_PACKAGE,
+            BuildConfig.APPLICATION_ID)
+        .append(AmplitudeEventLogger.VERSION_CODE, BuildConfig.VERSION_CODE)
+        .append(AmplitudeEventLogger.ENTRY_POINT,
+            if (entryPoint.isEmpty()) "other" else entryPoint)
+        .append(AmplitudeEventLogger.HAS_GMS, hasGms())
+
+    amplitudeClient.identify(identify)
+  }
+
+  fun start() {
+    idsRepository.getInstallerPackage(BuildConfig.APPLICATION_ID)
+        .doOnSuccess { installerPackage ->
+          val userId = idsRepository.getActiveWalletAddress()
+          val userLevel = idsRepository.getGamificationLevel()
+          amplitudeClient.initialize(context, BuildConfig.AMPLITUDE_API_KEY, userId)
+              .setTrackingOptions(TrackingOptions().disableAdid())
+          setAmplitudeSuperProperties(installerPackage, userLevel)
+        }
+        .subscribeOn(Schedulers.io())
+        .subscribe()
```

Comment:
Use .subscribe({},{it.printstacktrace}) to not break the app in case of an error